CustomToastStyle

joshuabourke opened the pull request:
feat: Allow users to customize toast message appearance

Users can now customize the appearance of toast messages.

A simple global toast style can be declared on the root view for convenience. Alternatively, users can override the global style for specific toasts by passing a style directly to the `.toast()` modifier.

![image](https://github.com/user-attachments/assets/4260abbd-d4f2-46f1-a711-c1da34e9954d)

joshuabourke committed:
CustomToastStyle

feat: Allow users to customize toast message appearance

Users can now customize the appearance of toast messages.

A simple global toast style can be declared on the root view for convenience. Alternatively, users can override the global style for specific toasts by passing a style directly to the `.toast()` modifier.

JamesSedlacek commented on .DS_Store:
Oh maybe the git ignore needs to be updated? And these DS Stores can be removed

JamesSedlacek commented on Sources/Toast/Modifiers/ToastModifier.swift:
```diff
@@ -84,9 +118,21 @@ struct ToastModifier<TrailingView: View>: ViewModifier {
+                    .gesture(dragGesture)
+            }
+        } else {
+            EmptyView()
```
nit: EmptyView doesn't need to be included

JamesSedlacek commented on Sources/Toast/Protocols/ToastStyle.swift:
```diff
@@ -0,0 +1,63 @@
+//
+//  File.swift
```
```suggestion
//  ToastStyle.swift
```

JamesSedlacek commented on Sources/Toast/Views/ToastMessageView.swift:
```diff
@@ -9,16 +9,54 @@ import SwiftUI
+        .init(toast: toast, trailingView: AnyView(trailingView))
+    }
+    
+    private var DefaultBody: some View {
```
nit: prefer camel case

JamesSedlacek commented on Sources/Toast/Views/ToastMessageView.swift:
```diff
@@ -48,6 +86,64 @@ struct ToastMessageView<TrailingView: View>: View {
+        .frame(maxWidth: .infinity, alignment: .leading)
+        .padding(8)
+        .background(.ultraThinMaterial)
+        .cornerRadius(12)
```
nit: `cornerRadius` is deprecated.

Preferred:
```swift
.background(
    .ultraThinMaterial,
    in: .rect(cornerRadius: 12)
)
```

JamesSedlacek commented:
This is pretty good! Would love to see some documentation in a README, so other people can read about how to use this.

joshuabourke committed:
Removed tracked .DS_Store files

joshuabourke committed:
ToastStyle PR Fixes

Cleanup and requested improvements:

• Removed DS_Store files
• Updated .gitignore to include .DS_Store
• Removed unnecessary EmptyView()
• Updated comment at top of ToastStyle file from "File" to "ToastStyle"
• Changed code to prefer camel casing
• Removed deprecated .cornerRadius modifier
• Updated README to show how to use ToastStyle

joshuabourke commented:
Implemented the changes you suggested and I also updated the ReadMe

joshuabourke committed:
Removed Duplicate Example

Removed the "PurpleToastStyle" example as this was just a duplicate of the "ExampleToastStyle"

JamesSedlacek commented:
Thanks for the contribution!
